lib/std/json.zig

```diff
@@ -1623,6 +1710,9 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                     }
                 }
             }
+            var buf: [structInfo.fields.len * (1 << 9)]u8 = undefined;
```

Comment:
I don't think the limit of 512 is appropriate; I was more thinking something like iterating over the fields (with an `inline for`) and calculating the actual max size of the hash table possible (if there is e.g. 5 fields, then the hash table can't have more than 5 entries).

Reply:
ok thats what i was thinking too initially.  but i'm not sure how to do it.  can you provide an example of how to initialize a hash table with only N entries?

Reply:
i think i found a solution. will push a commit soon.